From 475be6b868e08dbcb49f5730f7b2e6332a37e3d4 Mon Sep 17 00:00:00 2001
From: Matti Kosola <matti.kosola@jolla.com>
Date: Wed, 4 Sep 2019 12:07:34 +0200
Subject: [PATCH 39/41] (hybris) Disable SELinux init.

Sailfish OS SELinux initialization is done by systemd.
Conflicting init has been resolved by disabling SELinux
initialization from droid hal init.

Change-Id: I8a2a729671d224c4cc266a8219bee357611ded51
Signed-off-by: Matti Kosola <matti.kosola@jolla.com>
---
 init/init.cpp | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/init/init.cpp b/init/init.cpp
index 7fe36d504..ce2100c57 100644
--- a/init/init.cpp
+++ b/init/init.cpp
@@ -70,6 +70,8 @@ using android::base::StringPrintf;
 using android::base::Timer;
 using android::base::Trim;
 
+#define SYSTEMD_SELINUX
+
 namespace android {
 namespace init {
 
@@ -588,7 +590,9 @@ int main(int argc, char** argv) {
         gid_t groups[] = { AID_READPROC };
         setgroups(arraysize(groups), groups);
         //mount("sysfs", "/sys", "sysfs", 0, NULL);
+#ifndef SYSTEMD_SELINUX
         mount("selinuxfs", "/sys/fs/selinux", "selinuxfs", 0, NULL);
+#endif
 
         mknod("/dev/kmsg", S_IFCHR | 0600, makedev(1, 11));
 
@@ -622,6 +626,7 @@ int main(int argc, char** argv) {
         // Enable seccomp if global boot option was passed (otherwise it is enabled in zygote).
         global_seccomp();
 
+#ifndef SYSTEMD_SELINUX
         // Set up SELinux, loading the SELinux policy.
         SelinuxSetupKernelLogging();
         SelinuxInitialize();
@@ -631,6 +636,7 @@ int main(int argc, char** argv) {
         if (selinux_android_restorecon("/init", 0) == -1) {
             PLOG(FATAL) << "restorecon failed of /init failed";
         }
+#endif
 
         setenv("INIT_SECOND_STAGE", "true", 1);
 
@@ -687,10 +693,12 @@ int main(int argc, char** argv) {
     unsetenv("INIT_SELINUX_TOOK");
     unsetenv("INIT_AVB_VERSION");
 
+#ifndef SYSTEMD_SELINUX
     // Now set up SELinux for second stage.
     SelinuxSetupKernelLogging();
     SelabelInitialize();
     SelinuxRestoreContext();
+#endif
 
     epoll_fd = epoll_create1(EPOLL_CLOEXEC);
     if (epoll_fd == -1) {
-- 
2.20.1

